Services/UI/OJS.Services.Ui.Business/Implementations/SubmissionsHelper.cs

```diff
@@ -5,8 +5,13 @@ namespace OJS.Services.Ui.Business.Implementations;
 
 public class SubmissionsHelper : ISubmissionsHelper
 {
+    private readonly ISubmissionsForProcessingBusinessService submissionsForProcessingBusinessService;
+
+    public SubmissionsHelper(ISubmissionsForProcessingBusinessService submissionsForProcessingBusinessService)
+        => this.submissionsForProcessingBusinessService = submissionsForProcessingBusinessService;
     public bool IsEligibleForRetest(SubmissionDetailsServiceModel detailsModel)
         => (detailsModel.Tests.Any() && !detailsModel.TestRuns.Any()) &&
            detailsModel.IsProcessed && detailsModel.IsCompiledSuccessfully &&
-           (detailsModel.ProcessingComment == null);
+           (detailsModel.ProcessingComment == null) &&
+           this.submissionsForProcessingBusinessService.IsSubmissionProcessing(detailsModel.Id);
```

Comment:
Submission that is being processed should not be eligible for retest

Reply:
Fixed.